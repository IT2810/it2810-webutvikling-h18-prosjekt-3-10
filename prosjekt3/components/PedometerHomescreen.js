import React from "react";
import { Pedometer } from "expo";
import { Text } from 'react-native-elements';
import { AsyncStorage, StyleSheet, View } from "react-native";
import ProgressCircle from 'react-native-progress-circle'

export default class PedometerHomescreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            todayStepCount: 0,
            goal: '',
        }
    }

    componentDidMount() {
        this.retrieveData()
        this._subscribe();
    }

    /* componentDidUpdate() {
         this.retrieveData()
         this._subscribe();
     }*/

    retrieveData = async () => {
        try {
            const data = await AsyncStorage.getItem('Goal');
            if (data != null) {
                this.setState({
                    goal: data,
                });
            }
            else {
                this.setState({
                    goal: '',
                })
            }
        }
        catch (error) {
            alert('Error retrieving data')
        }
    }

    // function for step counter
    _subscribe = () => {
        this._subscription = Pedometer.watchStepCount(result => {
            this.setState({
                currentStepCount: result.steps
            });
        });
        const end = new Date();
        const start = new Date();
        start.setHours(end.getHours() - end.getHours());
        start.setMinutes(end.getMinutes() - end.getMinutes());
        Pedometer.getStepCountAsync(start, end).then(
            result => {
                this.setState({ todayStepCount: result.steps });
            },
            error => {
                this.setState({
                    pastStepCount: "Could not get stepCount: " + error
                });
            }
        );
    };

    // The progress circle
    progressCircle = () => {
        const steps = this.state.todayStepCount;
        const goal = this.state.goal;
        const perc = (parseInt(steps, 10) / parseInt(goal, 10) * 100);
        const empty = (this.state.goal === "0");
        return (
            <ProgressCircle
                percent={perc}
                radius={70}
                borderWidth={15}
                color='#e68a00'
                shadowColor="#ffffff"
                bgColor="#333333"
            >
                <Text style={styles.stepsText}> {steps} </Text>
                <Text style={styles.progressCircleText}>{empty ? "100%" : Math.ceil(100 * steps / goal) + "%"}</Text>
            </ProgressCircle>
        )
    }
    render() {
        return (
            <View >
                {this.progressCircle()}
            </View>
        )
    }
}

const styles = StyleSheet.create({

    progressCircleText: {
        fontSize: 18,
        color: "#999999"
    },

    stepsText: {
        fontSize: 30,
        fontWeight: 'bold',
        color: "#595959",
    },
})
