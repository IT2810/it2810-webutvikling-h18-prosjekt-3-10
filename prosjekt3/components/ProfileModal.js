import React from 'react';
import {
    StyleSheet,
    View,
    Modal,
    TouchableHighlight,
    Platform,
    SafeAreaView
} from 'react-native';
import { Icon, Header } from 'react-native-elements';
import ProfileScreen from '../screens/ProfilesScreen';
import Colors from '../constants/Colors'

export default class ProfileModal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalVisible: false,
        }
    }

    setModalVisible(visible) {
        this.setState({ modalVisible: visible });
    }

    ProfileMod = () => {
        return (
            <View>
                {/* the modal for profile */}
                <Modal
                    animationType="slide"
                    transparent={false}
                    visible={this.state.modalVisible}
                    presentationStyle="formSheet"
                    onRequestClose={() => { // required on android, making it posible to use hardware back-button
                        this.setModalVisible(!this.state.modalVisible);
                    }}>
                    <SafeAreaView style={styles.safeArea}>
                      <Header
                          outerContainerStyles={styles.headerOuterContainer}
                          innerContainerStyles={styles.headerInnerContainer}
                          rightComponent={{
                              size: 32,
                              icon: 'home',
                              color: Colors.btnBlue,
                              onPress: () => { this.setModalVisible(!this.state.modalVisible) }
                          }}
                          centerComponent={{
                              text: 'profile',
                              style: styles.header,
                          }}
                          backgroundColor={Colors.headerBackground}
                      />

                      <View >
                          {/* get return screen */}
                          <ProfileScreen />
                      </View>
                    </SafeAreaView>
                </Modal>

                <View style={styles.profileBtnContainer}>
                <TouchableHighlight
                    underlayColor={'transparent'}
                    onPress={() => { // opens modal when profile-object is pressed
                        this.setModalVisible(true);
                    }}>
                    <View>
                        {/* profile button on home screen */}
                        <Icon style={styles.modalProfile}
                            reverse
                            type='font-awesome'
                            name='user'
                            color='#2f95dc'
                            size={40}
                        />
                    </View>
                </TouchableHighlight>
                </View>
            </View>
        )
    }
    render() {
        return (
            <View>
                {this.ProfileMod()}
            </View>
        )
    }
}

const styles = StyleSheet.create({
    safeArea: {
      flex: 1,
      backgroundColor: '#fff',
    },
    profileBtnContainer: {
      flex: 1,
    },
    headerOuterContainer: {
        borderBottomWidth: 2,
        borderBottomColor: 'black',
    },
    header: {
        color: 'black',
        fontSize: 20,
        fontWeight: '600',
        fontVariant: ['small-caps'],
    },

})
