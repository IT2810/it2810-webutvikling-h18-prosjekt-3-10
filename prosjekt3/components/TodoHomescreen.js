import React from 'react';
import {
  StyleSheet,
  View,
  AsyncStorage,
  FlatList,
  ScrollView,
} from 'react-native';
import { List, ListItem } from 'react-native-elements';

export default class TodoHomescreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      todo: [],
      todayStepCount: 0,
    }
  }

  // mounting stored todo data
  componentDidMount() {
    this.retrieveData()
  }

  // updating todo between screens in the same session
  componentDidUpdate() {
    this.retrieveData()
  }

  // retrieving todo data from TodoScreen and displaying it on HomeScreen
  retrieveData = async () => {
    try {
      const getData = await AsyncStorage.getItem('Todo-list');
      const data = JSON.parse(getData);

      if (data != null) {
        this.setState({
          todo: data,
        });
      }
      else {
        this.setState({
          todo: [],
        })
      }
    }
    catch (error) {
      alert('Error retrieving data')
    }
  }

  // output from todo screen
  itemsOutput = () => {
    return (
      <List containerStyle={styles.todoList}>
        <ScrollView>
          <FlatList containerStyle={styles.todoList}
            data={this.state.todo}
            keyExtractor={(item) => item.id.toString()}
            extraData={this.state.refresh}
            renderItem={({ item }) => (
              <ListItem
                title={item.input}
                subtitle={item.date}
                hideChevron={true}
                style={{ backgroundColor: "blue" }}
              />
            )} />
        </ScrollView>
      </List>
    )
  }

  render() {
    return (
      <View >
        {this.itemsOutput()}
      </View>
    )
  }
}

const styles = StyleSheet.create({

  todoList: {
    borderWidth: 1,
    marginBottom: 10,
    marginTop: 0,
    width: 250,
    maxHeight: 120,
    minHeight: 100,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
  },
})

