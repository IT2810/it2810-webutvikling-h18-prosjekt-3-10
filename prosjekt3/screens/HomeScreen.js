import React from 'react';
import {
  StyleSheet,
  View,
  Text,
  ScrollView,
} from 'react-native';
import PedometerHomescreen from '../components/PedometerHomescreen';
import TodoHomescreen from '../components/TodoHomescreen';
import AppointmentModalHomescreen from '../components/AppointmentInfoHomescreenModal';
import ProfileModal from '../components/ProfileModal';
import { Header, } from 'react-native-elements';
import Colors from '../constants/Colors';

export default class HomeScreen extends React.Component {

  // header and styling of it
  static navigationOptions = {
    header: null, // removing the header
  };

  setModalVisible(visible) {
    this.setState({ modalVisible: visible });
  }

  render() {
    return (
      <ScrollView style={{
        backgroundColor: "#d9d9d9",
      }}>

        <View>

          <Header
            outerContainerStyles={styles.headerOuterContainer}
            innerContainerStyles={styles.headerInnerContainer}
            centerComponent={{
              text: 'STE reactive',
              style: styles.header,
            }}
            backgroundColor={Colors.headerBackground}
          />
          <View style={styles.modals}>

            {/* "upcoming events modal */}
            <AppointmentModalHomescreen />
            {/* button to profile modal*/}
            <ProfileModal />
          </View>

          {/* activity and todo content, middle of the screen */}
          <View style={styles.activityBackgroundContent}>

            {/* progress cycle*/}
            <View style={styles.activityContent}>
              <View>
                <PedometerHomescreen />
              </View>
            </View>
          </View>

          <View style={styles.TodoBackgroundContent}>

            {/* ToDo view content*/}
            <View style={styles.todoContent}>
              <Text style={styles.todoHeader}>ToDo</Text>
              <View>
                <TodoHomescreen />
              </View>
            </View>
          </View>
        </View>

      </ScrollView >
    );
  }
}

const styles = StyleSheet.create({

  activityContent: {
    padding: 10,
  },

  activityBackgroundContent: {
    backgroundColor: "#a6a6a6",
    borderColor: "#ffffff",
    borderBottomWidth: 1,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  activityHeader: {
    textAlign: "center",
    color: "#ffffff",
    fontSize: 16,
    marginBottom: 15,
  },

  headerInnerContainer: {
    marginTop: 15,
  },

  headerOuterContainer: {
    borderBottomWidth: 1,
    borderBottomColor: '#999',
    backgroundColor: "#5aa0dd",
    height: 100
  },

  header: {
    color: '#cccccc',
    fontSize: 25,
    fontWeight: '900',
    fontVariant: ['small-caps'],
  },

  TodoBackgroundContent: {
    backgroundColor: "#737373",
    borderColor: "#ffffff",
    borderBottomWidth: 1,
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },

  modals: {
    flexDirection: "row",
    backgroundColor: "#d9d9d9",
    borderColor: "#ffffff",
    borderBottomWidth: 1,
    borderTopWidth: 1,
  },

  todoContent: {
    padding: 10,
  },

  todoHeader: {
    textAlign: "center",
    color: "#ffffff",
    fontSize: 17,
    backgroundColor: "#425b84",
    borderColor: "#a6a6a6",
    borderRightWidth: 1,
    borderLeftWidth: 1,
    borderTopWidth: 1,
  },

});
