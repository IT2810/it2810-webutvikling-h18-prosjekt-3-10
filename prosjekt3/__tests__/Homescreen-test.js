import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../screens/HomeScreen';
import PedometerHomescreen from '../components/PedometerHomescreen';
import TodoHomescreen from '../components/TodoHomescreen';
import AppointmentModalHomescreen from '../components/AppointmentInfoHomescreenModal';
import ProfileModal from '../components/ProfileModal';

//testing snapshot view
// test('renders', () => {
//     const tree = renderer
//         .create(<PedometerHomescreen />)
//         .toJSON();
//     expect(tree).toMatchSnapshot();
// });

// component is rendered correctly
// test('renders button with passed props', () => {
//     const component = renderer.create(
//         <HomeScreen onClick={() => { }} label="test label" />
//     );
//     expect(component.toJSON()).toMatchSnapshot();
// });


// test('shallow test', () => {

//     const renderer = new ShallowRenderer(); renderer.render(<HomeScreen />);
//     const result = renderer.getRenderOutput();
//     expect(result.type).toBe("ImageBackground"); expect(result.props.children).toEqual([
//         <PedometerHomescreen />]);
// })


// const getDefaultProps = () => ({});
// describe('Risky component', () => {
//     it('shallow renders without crashing', () => {
//         const { } = getDefaultProps();
//         shallow(<HomeScreen />);
//     });
//     it('render snapshot', () => {
//         const { } = getDefaultProps();
//         const wrapper = render(<HomeScreen />);
//         expect(wrapper).toMatchSnapshot();
//     });
// });

