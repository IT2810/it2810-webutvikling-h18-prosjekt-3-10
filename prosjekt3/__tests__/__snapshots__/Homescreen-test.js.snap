// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homescreen snapshot renders the homes screen 1`] = `null`;
